Nodejs/Product/Nodejs/NpmUI/NpmWorker.cs

```diff
@@ -88,9 +88,9 @@ private void Run()
                 // The Take method will block the worker thread when there are no items left in the queue
                 // and the thread will be signalled when new items are items to the queue, or the queue is
                 // marked completed.
-                this.currentCommand = this.commandQueue.Take();
-                if (this.currentCommand != null)
+                if (this.commandQueue.TryTake(out var command, Timeout.Infinite) && command != null)
```

Comment:
`Take` throws an exception when the collection is marked as completed, where `TryTake` returns false. There was a racecondition I saw a couple of times between `Dispose` and `Take` being called.